File: TwoPointer/14.longest-common-prefix.go
package leetcode

import "slices"

// @leet start
func longestCommonPrefix(strs []string) string {
	if len(strs) == 0 {
		return strs[0]
	}
	slices.Sort(strs)
	l, r := strs[0], strs[len(strs)-1]
	for i := range len(l) {
		if l[i] != r[i] {
			return l[0:i]
		}
	}
	return l
}

// @leet end

File: BinarySearch/14.longest-common-prefix.py
from typing import List


# @leet start
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        strs.sort()
        lastItem, first = strs[-1], strs[0]
        i = 0
        for i in range(len(first)):
            if lastItem[i] != first[i]:
                # break # NOTE: is possible all match
                return first[:i]
        return first


# @leet end
